### PR TITLE
Snapshot failures should notify Slack

### PR DESCRIPTION
**Pull request by quazi-broad**
## Why
Snapshot failure does not notify slack
[Relevant ticket](https://broadinstitute.atlassian.net/browse/dspdc-1845)

## This PR
Added branching logic to return "NA" for job_id on a failed snapshot attempt

## Checklist
- [ ] Documentation has been updated as needed.

### CODE REVIEW

File: orchestration/dagster_orchestration/hca_orchestration/pipelines/cut_snapshot.py [hunk 1]
```diff
@@ -136,15 +136,19 @@ def snapshot_start_notification(context: HookContext) -> None:
     required_resource_keys={'slack', 'snapshot_config', 'dagit_config'}
 )
 def snapshot_job_failed_notification(context: HookContext) -> None:
-    job_id = context.solid_output_values["result"]
+    if "result" in context.solid_output_values:
+        job_id = context.solid_output_values["result"]
+    else:
+        job_id = "NA"
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_orchestration/pipelines/cut_snapshot.py:142]: ⛏️ _very_ small nitpick: `N/A` is more [conventional](https://en.wikipedia.org/wiki/N/A) for "not applicable"

File: orchestration/dagster_orchestration/hca_orchestration/pipelines/cut_snapshot.py [hunk 2]
```diff
@@ -136,15 +136,19 @@ def snapshot_start_notification(context: HookContext) -> None:
...
+
     kvs = {
         "Snapshot name": context.resources.snapshot_config.snapshot_name,
         "Dataset": context.resources.snapshot_config.dataset_name,
         "TDR Job ID": job_id,
         "Dagit link": f'<{context.resources.dagit_config.run_url(context.run_id)}|View in Dagit>'
     }
 
-    context.resources.slack.send_message(blocks=_base_slack_blocks("HCA Snapshot Failed", kvs))
+    context.resources.slack.send_message(blocks=_base_slack_blocks("HCA Snapshot Failed", key_values=kvs))
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_orchestration/pipelines/cut_snapshot.py:151]: 🔧 Double check the parameter name here (`key_values=`);  it looks like the second named parameter is `blocks` https://github.com/broadinstitute/dagster-utils/blob/4a17f7cb7d82ac188cc83cfdae7937f8b2804d94/dagster_utils/resources/slack.py#L18-L33

**Review by aherbst-broad — CHANGES_REQUESTED** [review]
Looks good, just need to double check the `key_values=` parameter naming.

File: orchestration/dagster_orchestration/hca_orchestration/pipelines/cut_snapshot.py (hunk 2 above)
Comment (quazi-broad) [reply orchestration/dagster_orchestration/hca_orchestration/pipelines/cut_snapshot.py:151]: I think this one might be okay - we are actually just passing in a single arg (in this case it's **blocks**) to send_message. The two "HCA Snapshot Failed" and kvs arguments are actually for that internal [_base_slack_blocks](https://github.com/DataBiosphere/hca-ingest/pull/159/files#diff-82a7e85b939a619e1117d426539fbb264a7e0f236cb26b165c6848fc85842c3bR90) function (line 90).

File: orchestration/dagster_orchestration/hca_orchestration/pipelines/cut_snapshot.py (hunk 1 above)
Comment (quazi-broad) [reply orchestration/dagster_orchestration/hca_orchestration/pipelines/cut_snapshot.py]: fixed! thank you

File: orchestration/dagster_orchestration/hca_orchestration/pipelines/cut_snapshot.py (hunk 2 above)
Comment (quazi-broad) [reply orchestration/dagster_orchestration/hca_orchestration/pipelines/cut_snapshot.py:151]: The change should have no effect anyways but I updated it to be consistent with line 132 when I was first looking for differences

**Review by aherbst-broad — APPROVED** [review]

**Review by eparkmonster — APPROVED** [review]

**Pull request merged**